### PR TITLE
[runtime env] Add environment variable to skip parsing `.gitignore` file into `"excludes"`

### LINKED ISSUE #18279: [runtime env] add "includes" or some other way of overriding "excludes"
### Describe your feature request
Currently runtime environments allows files to be excluded from being uploaded as part of the working_dir via the option `"excludes"`.  However, currently there's no way to override this force a file to be included.  

One concrete situation is if someone's project contains a `.gitignore` file containing some `.pkl` files.  (The `.pkl` files are necessary to run the code, but the user doesn't want them in their git repository.)  However, currently Ray parses the `.gitignore` file and excludes the listed files from being uploaded to the working directory.  The result is that the `.pkl` files are not uploaded to the cluster, and the user's code doesn't work.

For questions, post on our Discussion page instead: https://discuss.ray.io/

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Currently, if a user specifies a local `working_dir`, any `.gitignore` file in the directory will be parsed and the specified files will be excluded from the uploaded `working_dir` package.  This is an issue if the user wants to have some files in `.gitignore` (e.g. because they are very large) but still wants to upload them as part of the `working_dir`.  Previously, there was no way to override this behavior.

This PR adds an environment variable `RAY_RUNTIME_ENV_IGNORE_GITIGNORE` which if set to `1`, skips all parsing of `gitignore` files.  

## Related issue number

Closes #18279 
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
@dmatrix I'm assigning you for docs codeowner review since you have some context on the `runtime_env` docs, but feel free to reassign to another codeowner!

**Comment by architkulkarni** [comment]
(cc @gvspraveen )

**Review by edoakes — APPROVED** [review]
super nit: `IGNORE_GITIGNORE` is more fitting to me than `SKIP_GITIGNORE`

code changes LGTM

File: doc/source/ray-core/handling-dependencies.rst [hunk 1]
```diff
@@ -304,7 +304,7 @@ The ``runtime_env`` is a Python dictionary or a Python class :class:`ray.runtime
 
   Note: Setting a local directory per-task or per-actor is currently unsupported; it can only be set per-job (i.e., in ``ray.init()``).
 
-  Note: If your local directory contains a ``.gitignore`` file, the files and paths specified therein will not be uploaded to the cluster.  This can be disabled by setting the environment variable `RAY_RUNTIME_ENV_SKIP_GITIGNORE=1` on the machine doing the uploading.
+  Note: If your local directory contains a ``.gitignore`` file, the files and paths specified therein will not be uploaded to the cluster.  This can be disabled by setting the environment variable `RAY_RUNTIME_ENV_IGNORE_GITIGNORE=1` on the machine doing the uploading.
```
Comment (angelinalg) [inline doc/source/ray-core/handling-dependencies.rst:307]:
```suggestion
  Note: If the local directory contains a ``.gitignore`` file, the files and paths specified there are not uploaded to the cluster.  You can disable this by setting the environment variable `RAY_RUNTIME_ENV_IGNORE_GITIGNORE=1` on the machine doing the uploading.
```

File: doc/source/ray-core/handling-dependencies.rst [hunk 2]
```diff
@@ -325,7 +325,7 @@ The ``runtime_env`` is a Python dictionary or a Python class :class:`ray.runtime
 
   Note: Setting options (1) and (3) per-task or per-actor is currently unsupported, it can only be set per-job (i.e., in ``ray.init()``).
 
-  Note: For option (1), if your local directory contains a ``.gitignore`` file, the files and paths specified therein will not be uploaded to the cluster.  This can be disabled by setting the environment variable `RAY_RUNTIME_ENV_SKIP_GITIGNORE=1` on the machine doing the uploading.
+  Note: For option (1), if your local directory contains a ``.gitignore`` file, the files and paths specified therein will not be uploaded to the cluster.  This can be disabled by setting the environment variable `RAY_RUNTIME_ENV_IGNORE_GITIGNORE=1` on the machine doing the uploading.
```
Comment (angelinalg) [inline doc/source/ray-core/handling-dependencies.rst:328]:
```suggestion
  Note: For option (1), if the local directory contains a ``.gitignore`` file, the files and paths specified there are not uploaded to the cluster.  You can disable this by setting the environment variable `RAY_RUNTIME_ENV_IGNORE_GITIGNORE=1` on the machine doing the uploading.
```

**Review by angelinalg — CHANGES_REQUESTED** [review]
Left a couple of copy edits. Otherwise,
LGTM

**Comment by architkulkarni** [comment]
@angelinalg Thanks, I incorporated them! Please take another look

**Review by angelinalg — APPROVED** [review]
LGTM

**Comment by architkulkarni** [comment]
Doc build failing on master, error unrelated: /ray/doc/source/ray-air/examples/index.rst:29: WARNING: unknown document: /ray-air/examples/serving
RLlib cartpole tests flaky on master, unrelated
Debug wheels and Arm64 linux wheels flaky on master with error "Error: Observed wheel commit () is not expected commit (bda5766defb095ff370271648b3e8589685e39ce). Aborting.", unrelated
Tune `test_commands` flaky on master, unrelated

**Pull request merged**